server/controllers/conversation.js

```diff
@@ -0,0 +1,142 @@
+const mongoose = require('mongoose');
+const asyncHandler = require('express-async-handler');
+const Conversation = require('../models/Conversation');
+const Message = require('../models/Message');
+const User = require('../models/User');
+
+exports.createConversation = asyncHandler(async (req, res, next) => {
+  const { from, to } = req.body;
+
+  try {
+    if (!from || !to) {
+      res.status(400).json({
+        error: 'from and to are required',
+      });
+      throw new Error('from and to are required');
+    }
+
+    const fromUser = await User.findOne({ email: from });
+    const toUser = await User.findOne({ email: to });
+    if (!fromUser || !toUser) {
+      res.status(400).json({
+        error: 'unknown user',
+      });
+      throw new Error('unknown user');
+    }
+
+    const conversation = await Conversation.create({
+      fromUser: fromUser.get('_id'),
+      toUser: toUser.get('_id'),
+    });
+
+    res.status(201).json({
+      success: true,
+      conversation,
+    });
+  } catch (error) {}
+});
+
+exports.getUserConversations = asyncHandler(async (req, res, next) => {
+  const { email } = req.body;
+  try {
+    if (!email) {
+      res.status(400).json({
+        error: 'email is required',
+      });
+      throw new Error('email is required');
+    }
+
+    const user = await User.findOne({ email });
+    if (!user) {
+      res.status(400).json({
+        error: 'unknown user',
+      });
+      throw new Error('unknown user');
+    }
+    const userId = user.get('_id');
+    const conversations = await Conversation.find({
```

Comment:
The above steps can be avoided. If we use the auth middleware, we can get the logged in user's id from the request object.

Note: You can manually login with the API client first (that way the cookies are stored in your browser) and then make requests to these endpoints

Reply:
Done